### PR TITLE
Bugfix traceaction ancient

### PR DESCRIPTION
**Pull request by freddi8e**
Bugfix traceaction ancient

### CODE REVIEW

File: core/rawdb/freezer.go [hunk 1]
```diff
@@ -160,6 +170,20 @@ func NewFreezer(datadir string, namespace string, readonly bool, maxTableSize ui
 	return freezer, nil
 }
 
+// openAdditionTable create table, it will auto create new files when it was first initialized
+func openAdditionTable(datadir, name string, readMeter, writeMeter metrics.Meter, sizeGauge metrics.Gauge, maxTableSize uint32, disableSnappy, readonly bool) (*freezerTable, error) {
+	if readonly {
+		f, err := newTable(datadir, name, readMeter, writeMeter, sizeGauge, maxTableSize, disableSnappy, false)
```
Comment (Nanyan) [inline core/rawdb/freezer.go:176]: The `f` was not used?

File: core/rawdb/freezer.go [hunk 2]
```diff
@@ -160,6 +170,20 @@ func NewFreezer(datadir string, namespace string, readonly bool, maxTableSize ui
...
+		if err != nil {
+			return nil, err
+		}
+		if err = f.Close(); err != nil {
+			return nil, err
+		}
+	}
+	return newTable(datadir, name, readMeter, writeMeter, sizeGauge, maxTableSize, disableSnappy, readonly)
```
Comment (Nanyan) [inline core/rawdb/freezer.go:184]: always run to this line?

File: core/rawdb/freezer.go [hunk 3]
```diff
@@ -126,7 +127,16 @@ func NewFreezer(datadir string, namespace string, readonly bool, maxTableSize ui
 
 	// Create the tables.
 	for name, disableSnappy := range tables {
-		table, err := newTable(datadir, name, readMeter, writeMeter, sizeGauge, maxTableSize, disableSnappy, readonly)
+		var (
+			table *freezerTable
+			err   error
+		)
+		if slices.Contains(additionTables, name) {
```
Comment (Nanyan) [inline core/rawdb/freezer.go:134]: when will the new created table be added to `additionTables`

**Review by Nanyan — CHANGES_REQUESTED** [review]

File: core/rawdb/freezer.go (hunk 3 above)
Comment (freddi8e) [reply core/rawdb/freezer.go:134]: internaltx, the head of intrnaltx freezable MUST align with other freeze tables, such as block,recient ...
there no internal history records in ancientdb, so some extra check is needed.

File: core/rawdb/freezer.go (hunk 2 above)
Comment (freddi8e) [reply core/rawdb/freezer.go:184]: will not run if readonly case returns error

File: core/rawdb/freezer.go (hunk 1 above)
Comment (freddi8e) [reply core/rawdb/freezer.go:176]: line 188, f.close() , check if f.close works.

File: core/rawdb/freezer.go (hunk 3 above)
Comment (freddi8e) [reply core/rawdb/freezer.go:134]: hardcode, internaltx, 
new freeze table need some extra check,

**Pull request merged**